src/controllers/user.controller.ts

```diff
@@ -0,0 +1,32 @@
+import { NextFunction, Request, Response } from "express";
+import { User } from "../database/models";
+import bcrypt from "bcrypt";
+import config from "../config/config";
+import { HttpError } from "../errors/http.error";
+import httpStatus from "http-status";
+
+export const createUser = async (req: Request, res: Response, next: NextFunction) => {
+    
+    const {firstName, lastName, email, password} = req.body;
+
+    const userExist = await User.count({
+        where: {
+            email,
+        }
+    });
```

Comment:
@zaheerbabarkhan Instead of directly using the User model in the controller, it would be better to abstract these interactions into a separate service layer.

Reply:
oky, will move it to service layer